src/query/execute_statement.c

```diff
@@ -18712,6 +19381,91 @@ do_find_trigger_by_query (const char *name, char *buf, int buf_size)
 #undef QUERY_BUF_SIZE
 }
 
+int
+do_find_synonym_by_query (const char *name, char *buf, int buf_size)
+{
+#define QUERY_BUF_SIZE 2048
+  DB_QUERY_RESULT *query_result = NULL;
+  DB_QUERY_ERROR query_error;
+  DB_VALUE value;
+  const char *query = NULL;
+  char query_buf[QUERY_BUF_SIZE] = { '\0' };
+  int error = NO_ERROR;
+
+  db_make_null (&value);
+  query_error.err_lineno = 0;
+  query_error.err_posno = 0;
+
+  if (name == NULL || name[0] == '\0')
+    {
+      ERROR_SET_WARNING (error, ER_OBJ_INVALID_ARGUMENTS);
+      return error;
+    }
+
+  assert (buf != NULL);
+
+  query = "SELECT LOWER ([target_owner].[name] || '.' || [target_name]) FROM [%s] WHERE [unique_name] = '%s'";
```

Comment:
I know the LOWER function isn't perfect for collation. Is there any problem?

Reply:
I'll use target_unique_name .